File: apies/fake_data/src/models/Location.php
<?php

namespace atelier\fakedata\models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Concerns\HasUuids;

class Location extends Model
{
  use HasUuids;

  protected $connection = 'tedyspo_db';

  protected $table = 'location';
  protected $primaryKey = 'id_location';
  protected $autoIncrement = false;
  protected $keyType = 'string';
  public $timestamps = false;
  protected $fillable = [
    'id_location',
    'name',
    'lat',
    'long',
    'is_related',
    'id_event'
  ];

  public function event()
  {
    return $this->belongsTo(Event::class, 'id_event');
  }
}

File: flutter/lib/class/location.dart
class Location {
final String idLocation;
final String name;
final double lat;
final double long;
final bool isRelated;

Location({
required this.idLocation,
required this.name,
required this.lat,
required this.long,
required this.isRelated,
});

factory Location.fromMap(Map map) {
return Location(
idLocation: map['id'],
name: map['name'],
lat: map['lat'],
long: map['long'],
isRelated: map['is_related'] == 1 ? true : false,
);
}
}